src/test/java/core/basesyntax/service/TransactionServiceTest.java

```diff
@@ -0,0 +1,144 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.db.StorageDao;
+import core.basesyntax.service.impl.TransactionServiceImpl;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.transactions.FruitTransaction;
+import core.basesyntax.strategy.transactions.TransactionHandler;
+import core.basesyntax.strategy.transactions.impl.AdderHandler;
+import core.basesyntax.strategy.transactions.impl.ReduceHandler;
+import core.basesyntax.strategy.transactions.impl.SaverHandler;
+import java.util.HashMap;
+import java.util.List;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransactionServiceTest {
+    private static TransactionService transactionService;
+    private static StorageDao storageDao;
+
+    @BeforeClass
+    public static void beforeAll() {
+        storageDao = new StorageDao();
+        transactionService = new TransactionServiceImpl(getTransactionMap(storageDao));
+    }
+
+    @After
+    public void afterEach() {
+        Storage.getFruitStorage().clear();
+    }
+
+    @Test
+    public void transaction_applyPurchaseLessZeroPlus_Ok() {
+        storageDao.add("banana", 10);
+        transactionService.applyTransactions(List.of(
+                getFruitTransaction(Operation.PURCHASE, "banana", -10)
+        ));
+        Integer expected = 20;
+        Integer actual = storageDao.getValue("banana");
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void transaction_applyBalanceLessZero_ok() {
+        storageDao.add("banana", 0);
+        transactionService.applyTransactions(List.of(
+                getFruitTransaction(Operation.PURCHASE, "banana", 10)
+        ));
+        Integer expected = -10;
+        Integer actual = storageDao.getValue("banana");

```

Comment:
do you think the situation in storage when the quantity of fruits is less than 0 is ok?